experimental/dds/tree2/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -467,13 +505,28 @@ export function applyTypesFromContext(
 		const fields: Map<FieldKey, MapTree[]> = new Map();
 		for (const key of fieldKeysFromData(data)) {
 			assert(!fields.has(key), 0x6b3 /* Keys should not be duplicated */);
-			const childSchema = getFieldSchema(key, schemaData, schema);
-			const children = applyFieldTypesFromContext(schemaData, childSchema, data[key]);
+			const childSchema = getFieldSchema(key, context.schema, schema);
+			const children = applyFieldTypesFromContext(context, childSchema, data[key]);
+
 			if (children.length > 0) {
 				fields.set(key, children);
 			}
 		}
 
+		// Generate and set data for global keys that have value fields.
+		for (const [key] of context.schema.globalFieldSchema) {
+			const currentKey = symbolFromKey(key);
+			const requiredFieldSchema = getFieldSchema(currentKey, context.schema);
+			const multiplicity = getFieldKind(requiredFieldSchema).multiplicity;
+			if (multiplicity === Multiplicity.Value && context.fieldSource) {
```

Comment:
```suggestion
			if (multiplicity === Multiplicity.Value && context.fieldSource !== undefined) {
```
In this codebase we've chosen to prefer explicit undefined-ness checks over truthiness checks, even when it's equivalent.